### PR TITLE
Feature: Added GetUnitFlags(), GetUnitFlagsTwo(), SetUnitFlags(flags), SetUnitFlagsTwo(flags), Added PlayerEvent OnApplyAura/OnRemoveAura

### PR DESCRIPTION
**Pull request by New-HavenWotLK**
Implemented the stated methods to the source... tested them... they worked fine! =D

#138

### CODE REVIEW

**Comment by New-HavenWotLK** [comment]
@Helias if u got some time, could u check it out pls? =D

thx in advance! ^.^

**Comment by New-HavenWotLK** [comment]
added also now 2 new hooks which i have tested as well =)

they worked fine when checking the hooks via a print statement =D

**Comment by Helias** [comment]
thanks for your PR!
If the pipeline will pass successfully I will merge this PR

**Comment by Helias** [comment]
error in the build
```
/home/runner/work/mod-eluna/mod-eluna/modules/mod-eluna/src/LuaEngine/PlayerHooks.cpp:694:13: fatal error: out-of-line definition of 'OnApplyAura' does not match any declaration in 'Eluna'
void Eluna::OnApplyAura(Player* player, Aura* aura, bool isNewAura)
            ^~~~~~~~~~~
1 error generated.
make[2]: *** [modules/CMakeFiles/modules.dir/build.make:328: modules/CMakeFiles/modules.dir/mod-eluna/src/LuaEngine/PlayerHooks.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1199: modules/CMakeFiles/modules.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

File: src/LuaEngine/CreatureMethods.h [hunk 1]
```diff
@@ -914,6 +914,31 @@ auto const& threatlist = creature->GetThreatMgr().GetThreatList();
...
     }
+    
+    /**
+    * Returns the [Creature]'s Unit flags.
+    *
+    * These are used to control whether the NPC is attackable or not as well as many other different things,
+    *
+    *
+    * @return [unit_flags] unitFlags
+    */
```
Comment (r-o-b-o-t-o) [inline src/LuaEngine/CreatureMethods.h:925]:
```suggestion
    /**
     * Returns the [Creature]'s Unit flags.
     *
     * These are used to control whether the NPC is attackable or not, among other things.
     *
     * @return [UnitFlags] unitFlags
     */
```

File: src/LuaEngine/CreatureMethods.h [hunk 2]
```diff
@@ -914,6 +914,31 @@ auto const& threatlist = creature->GetThreatMgr().GetThreatList();
...
+    {
+        Eluna::Push(L, creature->GetUInt32Value(UNIT_FIELD_FLAGS));
+        return 1;
+    }
+
+    /**
+    * Returns the [Creature]'s Unit flags 2.
+    *
+    * @return [unit_flags2] unitFlags2
+    */
```
Comment (r-o-b-o-t-o) [inline src/LuaEngine/CreatureMethods.h:936]:
```suggestion
    /**
     * Returns the [Creature]'s Unit flags 2.
     *
     * @return [UnitFlags2] unitFlags2
     */
```

File: src/LuaEngine/CreatureMethods.h [hunk 3]
```diff
@@ -923,6 +948,7 @@ auto const& threatlist = creature->GetThreatMgr().GetThreatList();
      *
      * @return [ExtraFlags] extraFlags
      */
+
```
Comment (r-o-b-o-t-o) [inline src/LuaEngine/CreatureMethods.h:951]:
```suggestion
```

File: src/LuaEngine/CreatureMethods.h [hunk 4]
```diff
@@ -982,6 +1008,30 @@ auto const& threatlist = creature->GetThreatMgr().GetThreatList();
         creature->SetUInt32Value(UNIT_NPC_FLAGS, flags);
         return 0;
     }
+    
+    /**
+    * Sets the [Creature]'s UNIT flags to `flags`.
+    *
+    * @param [unit_flags] flags
+    */
```
Comment (r-o-b-o-t-o) [inline src/LuaEngine/CreatureMethods.h:1016]:
```suggestion
    /**
     * Sets the [Creature]'s Unit flags to `flags`.
     *
     * @param [UnitFlags] flags
     */
```

File: src/LuaEngine/CreatureMethods.h [hunk 5]
```diff
@@ -982,6 +1008,30 @@ auto const& threatlist = creature->GetThreatMgr().GetThreatList();
...
+        uint32 flags = Eluna::CHECKVAL<uint32>(L, 2);
+        creature->SetUInt32Value(UNIT_FIELD_FLAGS, flags);
+        return 0;
+    }
+
+    /**
+    * Sets the [Creature]'s UNIT flags2 to `flags`.
+    *
+    * @param [unit_flags2] flags
+    */
```
Comment (r-o-b-o-t-o) [inline src/LuaEngine/CreatureMethods.h:1028]:
```suggestion
    /**
     * Sets the [Creature]'s Unit flags2 to `flags`.
     *
     * @param [UnitFlags2] flags
     */
```

**Review by r-o-b-o-t-o — CHANGES_REQUESTED** [review]
Hi there, thanks for the PR!

Looks like some parts are missing for your hooks, see this commit for an example on how to add a hook: https://github.com/azerothcore/mod-eluna/commit/85714c1123489ff8d110f50a4701b293acc95c43

Also, could you document your additions in `README.md` please? See https://github.com/azerothcore/mod-eluna/commit/2a683a59e37cf08f4912aac2016467d99f412849 and the other commit I linked for examples.

**Comment by New-HavenWotLK** [comment]
ah uups... XD

i will look to fix this when i am home tomorrow =)

**Comment by New-HavenWotLK** [comment]
sooo i think now there should be all properly added

shouldn't add stuff by hand when i already am half asleep ^^'

**Comment by Helias** [comment]
@r-o-b-o-t-o what do you think?

**Review by r-o-b-o-t-o — APPROVED** [review]
Looks good, thanks!

**Comment by r-o-b-o-t-o** [comment]
Just noticed this now while looking into https://github.com/azerothcore/mod-eluna/issues/144, you're using hooks that don't exist for your aura events...
I didn't notice the missing `override`s in `ElunaLuaEngine_SC.cpp` when reviewing.
Since the real aura hooks are in `UnitScript` and not `PlayerScript` I'm afraid I will have to revert this part of the PR, since we don't have Unit hooks.

**Pull request merged**